src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -8,11 +8,14 @@
 namespace System.Collections.Generic
 {
     /// <summary>
-    /// Represents a data structure in which each element has an associated priority
-    /// that determines the order in which the pair is dequeued.
+    ///  Represents a min priority queue.
     /// </summary>
-    /// <typeparam name="TElement">The type of the element.</typeparam>
-    /// <typeparam name="TPriority">The type of the priority.</typeparam>
+    /// <typeparam name="TElement">Specifies the type of elements in the queue.</typeparam>
+    /// <typeparam name="TPriority">Specifies the type of priority associated with enqueued elements.</typeparam>
+    /// <remarks>
+    ///  Implements an array-backed quaternary min-heap. Each element is enqueued with an associated priority
```

Comment:
The array-backed and quarternary parts of this are implementation detail.  Do they need to be in here?  Are we guaranteeing that will be the case forever more?

Reply:
> Are we guaranteeing that will be the case forever more?

I'm thinking that this is more informational rather than a behaviour users would rely on. As such, we could always update the docs as we evolve the implementation without consequence.